tests/integration/fixtures/processing.py

```diff
@@ -118,7 +118,7 @@ def inner(topic: str, options=None):
 
         settings = {
             "bootstrap.servers": servers,
-            "group.id": "test.consumer",
+            "group.id": "test.consumer.%s" % uuid.uuid4(),
```

Comment:
This is a mere mitigation, but it seems to be fairly effective. By creating a new consumer group for every consumer, we ensure that every message is delivered to every consumer should they ever share the same topic.

I have no idea how it can be that two consumers share the same topic, but my local tests indicate that this might be the case. Maybe it has to do with how xdist works?

Reply:
the topic name contains a random integer.

Reply:
> the topic name contains a random integer.

Sure, that's obvious. Still, I can only observer that when tests run concurrently, messages are consumed before the consumer attaches, which suggests to me that they leak across concurrent tests.